plugins/base/src/main/kotlin/transformers/pages/merger/SourceSetMergingPageTransformer.kt

```diff
@@ -0,0 +1,38 @@
+package org.jetbrains.dokka.base.transformers.pages.merger
+
+import org.jetbrains.dokka.Platform
+import org.jetbrains.dokka.model.ContentSourceSet
+import org.jetbrains.dokka.model.toContentSourceSets
+import org.jetbrains.dokka.pages.*
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.transformers.pages.PageTransformer
+
+class SourceSetMergingPageTransformer(context: DokkaContext) : PageTransformer {
+
+    private val mergedSourceSets = context.configuration.sourceSets.toContentSourceSets()
+        .associateBy { sourceSet -> sourceSet.key }
+
+    override fun invoke(input: RootPageNode): RootPageNode {
+        return input.transformContentPagesTree { contentPage ->
+            val content: ContentNode = contentPage.content
+            val newContent = transformWithMergedSourceSets(content)
```

Comment:
nit: imho you can inline it to: `val newContent = transformWithMergedSourceSets(contentPage.content)`

Reply:
Good point ☺️ 
Done ✅